### PR TITLE
Add 'shuffle' and 'repeat' properties to the SoCo class

### PR DESCRIPTION
**Pull request by ghcs27**
To control repeat and shuffle, we currently have the `play_mode` property, which is a string combining both properties. To turn on shuffle, the a controller using SoCo has to interpret this string as a play mode, set the shuffle flag to True and reformat it to a string. To make this easier, I made `shuffle` and `repeat` properties which do this work. They use the `PLAY_MODES` dict, which maps each play mode string to a (shuffle, repeat) tuple.

### CODE REVIEW

File: soco/core.py [hunk 1]
```diff
@@ -1573,16 +1603,6 @@ def __get_favorites(self, favorite_type, start=0, max_items=100):
 
         return result
 
-    def _update_album_art_to_full_uri(self, item):
```
Comment (amelchio) [inline soco/core.py]: This cleanup seems unrelated.

Comment (ghcs27) [reply soco/core.py]: Thanks, I mixed up my branches.
